File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        for (int i = 0; i < nums.size(); i++)
        {
            int temp = nums[i];
            for (int k = i+1 ; k < nums.size() ; k++)
            {
                if(target == temp + nums[k])
                {
                    return {i, k};
                }
            }
        }
        //if we can't find anything just send an empty vector in 
        return {};
    }
};

File: 1-two-sum/two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
  const map = new Map();

  for (let i = 0; i < nums.length; i++) {
    const tmpNum = target - nums[i];
    if (!map.has(tmpNum)) {
      map.set(nums[i], i);
    } else {
      return [i, map.get(tmpNum)];
    }
  }
};
